pyproject.toml

```diff
@@ -166,54 +166,66 @@ select = [  # Base linting rule selections.
+    "PIE796",  # Duplicate value in an enum (reasonable to noqa for backwards compatibility)
+    "PIE804",  # Don't use a dict with unnecessary kwargs
+    "PIE807",  # prefer `list` over `lambda: []`
+    "PIE810",  # Use a tuple rather than multiple calls. E.g. `mystr.startswith(("Hi", "Hello"))`
     "PYI",  # Linting for type stubs.
     "PT",  # Pytest
     "Q",  # Consistent quotations
     "RSE",  # Errors on pytest raises.
     "RET",  # Simpler logic after return, raise, continue or break
+    "SLF",  # Error when accessing private members
     "SIM",  # Code simplification
+    "TID",  # Tidy imports
+    # The team have chosen to only use type-checking blocks when necessary to prevent circular imports.
+    # As such, the only enabled type-checking checks are those that warn of an import that needs to be
+    # removed from a type-checking block.
     "TCH004",  # Remove imports from type-checking guard blocks if used at runtime
     "TCH005",  # Delete empty type-checking blocks
     "ARG",  # Unused arguments
     "PTH",  # Migrate to pathlib
+    "FIX",  # All TODOs, FIXMEs, etc. should be turned into issues instead.
     "ERA",  # Don't check in commented out code
     "PGH",  # Pygrep hooks
     "PL",  # Pylint
     "TRY",  # Cleaner try/except,
-]
-extend-select = [
-    # Pyupgrade: https://github.com/charliermarsh/ruff#pyupgrade-up
-    "UP00", "UP01", "UP02", "UP030", "UP032", "UP033",
-    # "UP034",  # Very new, not yet enabled in ruff 0.0.227
-    # Annotations: https://github.com/charliermarsh/ruff#flake8-annotations-ann
-    "ANN0",  # Type annotations for arguments other than `self` and `cls`
-    "ANN2",  # Return type annotations
-    "B026",  # Keyword arguments must come after starred arguments
-    # flake8-bandit: security testing. https://github.com/charliermarsh/ruff#flake8-bandit-s
-    # https://bandit.readthedocs.io/en/latest/plugins/index.html#complete-test-plugin-listing
-    "S101", "S102",  # assert or exec
-    "S103", "S108",  # File permissions and tempfiles - use #noqa to silence when appropriate.
-    "S104",  # Network binds
-    "S105", "S106", "S107",  # Hardcoded passwords
-    "S110",  # try-except-pass (use contextlib.suppress instead)
-    "S113",  # Requests calls without timeouts
-    "S3",  # Serialising, deserialising, hashing, crypto, etc.
-    "S506",  # Unsafe YAML load
-    "S508", "S509",  # Insecure SNMP
-    "S701",  # jinja2 templates without autoescape
+    "FLY",  # Detect things that would be better as f-strings.
+    "PERF",  # Catch things that can slow down the application like unnecessary casts to list.
     "RUF001", "RUF002", "RUF003",  # Ambiguous unicode characters
     "RUF005",  # Encourages unpacking rather than concatenation
     "RUF008",  # Do not use mutable default values for dataclass attributes
+    "RUF011",  # Don't use static keys in dict comprehensions.
+    "RUF012",  # Mutable class attributes should be annotated with typing.ClassVar
```

Comment:
Follow RUF012 will broken `pydantic` that it cannot found the `ClassVar`

Reply:
Thanks for pointing this out. Removed!